pyannote/audio/features/utils.py

```diff
@@ -259,33 +255,40 @@ def crop(self, current_file, segment, mode='center', fixed=None):
         --------
         `pyannote.core.SlidingWindowFeature.crop`
         """
-
         if self.sample_rate is None:
             msg = ('`RawAudio` needs to be instantiated with an actual '
                    '`sample_rate` if one wants to use the `crop` method.')
             raise ValueError(msg)
 
+        # find the start and end positions of the required segment
+        (start, end), = self.sliding_window_.crop(
+            segment, mode=mode, fixed=fixed, return_ranges=True)
+
         if 'waveform' in current_file:
             y = current_file['waveform']
             sample_rate = self.sample_rate
+            data = y[start:end]
 
-        else:
-
-            warnings.filterwarnings("ignore", category=WavFileWarning)
-
-            # read data as memory mapped
-            sample_rate, y = scipy.io.wavfile.read(current_file['audio'],
-                                                   mmap=True)
-
-            if sample_rate != self.sample_rate:
-                msg = (f'Mismatch between expected ({self.sample_rate:d}) and '
-                       f'actual ({sample_rate} sample rates)')
+        else:  # read file with SoundFile, which supports various fomats, including NIST sphere
+            try:
+                with SoundFile(current_file['audio'], 'r') as audio_file:
+                    sample_rate = audio_file.samplerate
+                    # if the sample rates are mismatched, recompute the start and end
+                    if sample_rate != self.sample_rate:
+                        start = int(1. * sample_rate / self.sample_rate * start)
+                        end = int(1. * sample_rate / self.sample_rate * end)
+                    audio_file.seek(start)
+                    data = audio_file.read(end - start, dtype='float32')
+
+            except OSError as e:
+                msg = ('ERROR: problems when reading file {0} with segment {1}. '.format(current_file, segment) + str(e) )
                 raise ValueError(msg)
 
-        # extract segment waveform
-        (start, end), = self.sliding_window_.crop(
-            segment, mode=mode, fixed=fixed, return_ranges=True)
-        data = y[start:end]
+            # if sample rate of the file we just read does not match the expected one,
+            # resample the piece of data on the fly
+            if sample_rate != self.sample_rate:
+                data = librosa.core.resample(data, sample_rate, self.sample_rate)
+                sample_rate = self.sample_rate
```

Comment:
Here, we should check that `data` contains exactly the correct number of samples (as defined at line 264).

Reply:
Fixed in commit d16cae6